src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -16,20 +16,46 @@
 @CrossOrigin(origins = "http://localhost:3000")
 public class StudentController {
 
+    @Autowired
+    EnrollmentRepository enrollmentRepository;
+
+    @Autowired
+    SectionRepository sectionRepository;
+
+    @Autowired
+    UserRepository userRepository;
 
    // student gets transcript showing list of all enrollments
    // studentId will be temporary until Login security is implemented
    //example URL  /transcript?studentId=19803
    @GetMapping("/transcripts")
    public List<EnrollmentDTO> getTranscript(@RequestParam("studentId") int studentId) {
 
-       // TODO
-
-       // list course_id, sec_id, title, credit, grade in chronological order
-       // user must be a student
-	   // hint: use enrollment repository method findEnrollmentByStudentIdOrderByTermId
-       // remove the following line when done
-       return null;
+       List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsByStudentIdOrderByTermId(studentId);
+       List<EnrollmentDTO> enrollmentDTOs = new ArrayList<>();
+
+       for (Enrollment e : enrollments) {
+           EnrollmentDTO dto = new EnrollmentDTO(
+                   e.getEnrollmentId(),
+                   e.getGrade(),
+                   e.getUser().getId(),
+                   e.getUser().getName(),
```

Comment:
The instructor email, building, room, and times are optional parameters; the top comment asks to only list course_id, sec_id, title, credit, and grade.